lnbits/wallets/lndrest.py

```diff
@@ -154,29 +178,53 @@ async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse
             r.raise_for_status()
         except Exception as exc:
             logger.warning(f"LndRestWallet pay_invoice POST error: {exc}.")
-            return PaymentResponse(None, None, None, None, str(exc))
+            return PaymentResponse(
+                None, None, None, None, f"Unable to connect to {self.endpoint}."
+            )
 
-        data = r.json()
+        try:
+            data = r.json()
 
-        if data.get("payment_error"):
-            error_message = r.json().get("payment_error") or r.text
-            logger.warning(f"LndRestWallet pay_invoice payment_error: {error_message}.")
-            return PaymentResponse(False, None, None, None, error_message)
+            if data.get("payment_error"):
+                error_message = r.json().get("payment_error") or r.text
+                logger.warning(
+                    f"LndRestWallet pay_invoice payment_error: {error_message}."
+                )
+                return PaymentResponse(False, None, None, None, error_message)
+
+            if (
+                "payment_hash" not in data
+                or "payment_route" not in data
+                or "total_fees_msat" not in data["payment_route"]
+                or "payment_preimage" not in data
+            ):
+                return PaymentResponse(
+                    False, None, None, None, "Server error: 'missing required fields'"
+                )
 
-        data = r.json()
-        checking_id = base64.b64decode(data["payment_hash"]).hex()
-        fee_msat = int(data["payment_route"]["total_fees_msat"])
-        preimage = base64.b64decode(data["payment_preimage"]).hex()
-        return PaymentResponse(True, checking_id, fee_msat, preimage, None)
+            checking_id = base64.b64decode(data["payment_hash"]).hex()
+            fee_msat = int(data["payment_route"]["total_fees_msat"])
+            preimage = base64.b64decode(data["payment_preimage"]).hex()
+            return PaymentResponse(True, checking_id, fee_msat, preimage, None)
+        except json.JSONDecodeError:
+            return PaymentResponse(
+                False, None, None, None, "Server error: 'invalid json response'"
```

Comment:
can we use kwargs in places like that? aka `return PaytmentRepsonse(ok=False, error_msg="invalid...")`

Reply:
Yes, we can and we should, there is a `PaymentResponse[Success | Failed | Pending]` PR.

In this PR I have made minimum changes to the code, the focus is to introduce the tests so future changes are less risky.